### ISSUE TITLE
LORA微调internvl2模型时，出现大量 tokenization mismatch警告

### ISSUE THREAD

**Issue by feihuamantian** [OpenGVLab/InternVL#347]
[2024-07-10 17:42:44,683] [INFO] [logging.py:96:log_dist] [Rank 0] time (ms) | fwd_microstep: 998.17 | bwd_microstep: 1538.87 | bwd_inner_microstep: 1536.33 | bwd_allreduce_microstep: 2.47 | step_microstep: 0.06
WARNING: tokenization mismatch: 654 vs. 643. #turn = 25. (ignored). This dataset is docvqa_train_10k.
WARNING: tokenization mismatch: 737 vs. 726. #turn = 25. (ignored). This dataset is docvqa_train_10k.
WARNING: tokenization mismatch: 405 vs. 402. #turn = 9. (ignored). This dataset is docvqa_train_10k.
WARNING: tokenization mismatch: 327 vs. 326. #turn = 5. (ignored). This dataset is docvqa_train_10k.
WARNING: tokenization mismatch: 337 vs. 336. #turn = 5. (ignored). This dataset is docvqa_train_10k.
WARNING: tokenization mismatch: 660 vs. 649. #turn = 25. (ignored). This dataset is docvqa_train_10k.
WARNING: tokenization mismatch: 334 vs. 333. #turn = 5. (ignored). This dataset is docvqa_train_10k.
WARNING: tokenization mismatch: 485 vs. 480. #turn = 13. (ignored). This dataset is docvqa_train_10k.
[2024-07-10 17:42:45,900] [INFO] [logging.py:96:log_dist] [Rank 0] time (ms) | optimizer_step: 0.94
[2024-07-10 17:42:45,900] [INFO] [logging.py:96:log_dist] [Rank 0] time (ms) | fwd_microstep: 528.30 | bwd_microstep: 627.38 | bwd_inner_microstep: 624.77 | bwd_allreduce_microstep: 2.54 | step_microstep: 49.90
[2024-07-10 17:42:45,900] [INFO] [logging.py:96:log_dist] [Rank 0] time (ms) | fwd: 1526.44 | bwd: 2166.26 | bwd_inner: 2161.14 | bwd_allreduce: 5.02 | step: 49.96
{'loss': 1.4251, 'learning_rate': 9.667469644026118e-06, 'epoch': 0.17}
 17%|█▋        | 111/672 [06:39<33:56,  3.63s/it]WARNING: tokenization mismatch: 480 vs. 475. #turn = 13. (ignored). This dataset is docvqa_train_10k.
WARNING: tokenization mismatch: 515 vs. 508. #turn = 17. (ignored). This dataset is docvqa_train_10k.
WARNING: tokenization mismatch: 390 vs. 387. #turn = 9. (ignored). This dataset is docvqa_train_10k.
WARNING: tokenization mismatch: 436 vs. 431. #turn = 13. (ignored). This dataset is docvqa_train_10k.
WARNING: tokenization mismatch: 317 vs. 316. #turn = 5. (ignored). This dataset is docvqa_train_10k.
WARNING: tokenization mismatch: 330 vs. 329. #turn = 5. (ignored). This dataset is docvqa_train_10k.
WARNING: tokenization mismatch: 907 vs. 896. #turn = 25. (ignored). This dataset is docvqa_train_10k.
WARNING: tokenization mismatch: 1012 vs. 997. #turn = 33. (ignored). This dataset is docvqa_train_10k.

LORA微调时，出现大量tokenization mismatch，请问这种会影响微调结果吗

**Comment by HaoRenkk123** [comment]
I got the same warnings, need help

**Comment by HaoRenkk123** [comment]
https://github.com/OpenGVLab/InternVL/blob/main/internvl_chat/shell/phi3_3_8b_dynamic/internvl_chat_v1_5_phi3_3_8b_dynamic_res_finetune.sh
conv_style change to "phi3-chat"

**Comment by feihuamantian** [comment]
> phi3-chat
 thanks very much!

**Comment by feihuamantian** [comment]
> https://github.com/OpenGVLab/InternVL/blob/main/internvl_chat/shell/phi3_3_8b_dynamic/internvl_chat_v1_5_phi3_3_8b_dynamic_res_finetune.sh conv_style change to "phi3-chat"
您好，按照您说的，将conv_style改为“phi3-chat”,lora微调的时候还是会出现这个警告，不过比以前少了一些，不知道是否还是有问题？
![image](https://github.com/OpenGVLab/InternVL/assets/28010145/bf8bc89f-d5fc-46cf-bcc2-4cb585064593)

**Comment by middleflames** [comment]
Same problem here. I tried phi3-chat as well as the default one in the shell script, but the mismatch still exists.

**Comment by G-z-w** [comment]
Our new code has been released, please refer to https://internvl.readthedocs.io/en/latest/internvl2.0/finetune.html#start-2nd-fine-tuning.

**Comment by funny000** [comment]
Hi, @G-z-w
Is there any improvement in the effect after fine-tuning?

**Comment by G-z-w** [comment]
The effectiveness of fine-tuning depends largely on your task and training data.

**Comment by funny000** [comment]
Hi@G-z-w 
I see the readdoc ([https://internvl.readthedocs.io/en/latest/internvl2.0](https://internvl.readthedocs.io/en/latest/internvl2.0/finetune.html#start-2nd-fine-tuning)) and use coco to evaluate 
 internvl ,
the command line is GPUS=4 sh evaluate.sh pretrained/InternVL2-2B caption-coco --dynamic,
where is the script of evaluate.sh?

**Comment by czczup** [comment]
This script is located at here: https://xgithub.com/OpenGVLab/InternVL/blob/main/internvl_chat/evaluate.sh

**Comment by zhouxincheng** [comment]
> This script is located at here: https://xgithub.com/OpenGVLab/InternVL/blob/main/internvl_chat/evaluate.sh

The URL was misspelled, and it redirected to a gambling website. https://github.com/OpenGVLab/InternVL/blob/main/internvl_chat/evaluate.sh

**Comment by wangpeng138375** [comment]
多轮对话，当不是human, gpt 不成对出现，或者最后一个角色不是gpt等情况，都会出现这个问题
